feat: check pull requests and commits using conventional commits

ferdodo opened the pull request:
I added two new more workflows, the purpose is to have better consistency over the commit history and pull requests.

- The `./git-conventional-commits.yaml` configuration file is enforced on the latest commit of the pull request.
- Pull requests must respect a basic name and description regex based on conventional commit specification.
- Improved job and steps names for other workflows.

![image](https://github.com/DevGirl-Team/tamagokill/assets/2821066/d5070cc8-d64e-4467-bb28-60762678651b)

Jean-BaptisteC commented:
Maybe use latest version of checkout and setup-node actions?
checkout v3 use a deprecated version of node js.

ferdodo commented:
it is indeed using node16, I though it was supposed to take the node version from package.json as mentionned here https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file , but you actually have to reference where the package.json is at.

I'll add that too.

pbrissaud commented:
To avoid using a deprecated version, we can use a tool like [Renovate](https://www.mend.io/renovate/) to automatically create PRs when a dependency update (supports GitHub action, npm, and cargo) is available. I use it in my personal projects, and it works like a charm! And it's free for open-source projects!

pbrissaud commented on .github/workflows/check-code.yml:
```diff
@@ -29,11 +30,12 @@ jobs:
         working-directory: ./web
```
Why build the app after testing? I think it's useless

ferdodo replied:
it was intended for running typechecking, I guess I should run the "check" script instead.

pbrissaud commented:
I find checking both the PR title and all commits for adherence to conventional commits redundant, especially when using squash merges. Squash merging results in all changes combined into a single commit with the PR title, making individual commit checks unnecessary. Conversely, if using merge or rebase, the separate commit messages are visible, rendering the PR title check redundant. Simplifying the checks to align with the merge strategy would improve efficiency and avoid unnecessary duplication.

ferdodo commented:
@pbrissaud 
![image](https://github.com/DevGirl-Team/tamagokill/assets/2821066/9e781a91-bf05-42c2-9469-e1dbab68cfb5)
Some messages are squashed and others kept unchanged

pbrissaud commented:
> Some messages are squashed and others kept unchanged

Hmm, I didn't know that it's possible to have both at the same time 🤔 Would it not be better to define and enforce a single merging strategy?

ferdodo committed:
feat: check pull requests and commits using conventional commits

ferdodo commented:
I just updated,

- PR title is now checked against the configuration file.
- I removed the PR description check.
- the 'check' script is run instead of the 'build' script.
- setup-node shall run the node version specified in package.json

pbrissaud reviewed (approved):
LGTM
